src/sql/workbench/parts/dashboard/widgets/properties/propertiesWidget.component.ts

```diff
@@ -111,16 +139,104 @@ export class PropertiesWidgetComponent extends DashboardWidget implements IDashb
+		}
+		else if (this._config.widget['properties-widget'] && this._config.widget['properties-widget'].endpoints) {
+			this._isVertical = true;
+			this.parseEndpoints();
+		}
+		else {
+			this.parseProperties(null);
+		}
+		this._loading = false;
+	}
+
+	private parseEndpoints(): void {
+		if (!this.endpoints || this.endpoints.length === 0) {
+			this.getEndpoints();
+		}
+
+		// iterate over endpoint properties and display them
+		this.properties = [];
+		for (const endpoint of this.endpoints) {
+			let propertyObject = this.getValueOrDefault<string>(endpoint, EndpointConstants.hyperlink.toString(), '--');
+			const assignProperty: DisplayProperty = {
+				displayName: endpoint.serviceName,
+				value: propertyObject,
+				isHyperlink: this.isHyperlink(endpoint.serviceName)
+			};
+
+			this.properties.push(assignProperty);
+		}
+
+		if (this._hasInit) {
+			this._changeRef.detectChanges();
+		}
+	}
+
+	private getEndpoints(): void {
+		const endpointArray = this._connection.serverInfo.options[EndpointConstants.clusterEndpoints.toString()];
+
+		// TO DO: Once the endpoints data is moved to DMV, update this logic to replace hardcoded values
+		// with the data from DMV.
+		if (endpointArray && endpointArray.length > 0) {
+			// iterate over endpoints and display them
+			this.endpoints = [];
+			for (const ep of endpointArray) {
+				const assignProperty: Endpoint = {
+					serviceName: ep.serviceName,
+					ipAddress: ep.ipAddress,
+					port: ep.port,
+					hyperlink: ep.ipAddress + ':' + ep.port
+				};
+				this.endpoints.push(assignProperty);
+			}
+
+			// add grafana and kibana endpoints from the management proxy endpoint
+			const managementProxyEp = endpointArray.find(e => e.serviceName === EndpointConstants.managementproxy.toString());
+			if (!endpointArray.find(e => e.serviceName.toLowerCase().indexOf('metrics') > -1)) {
+				this.endpoints.push(this.addCustomEndpoint(managementProxyEp, 'Grafana Dashboard', '/grafana'));
+			}
+			if (!endpointArray.find(e => e.serviceName.toLowerCase().indexOf('log') > -1)) {
+				this.endpoints.push(this.addCustomEndpoint(managementProxyEp, 'Kibana Dashboard', '/kibana'));
+			}
+
+			// add spark and yarn endpoints form the gateway endpoint
+			const gatewayEndpoint = endpointArray.find(e => e.serviceName === EndpointConstants.gateway.toString());
+			if (!endpointArray.find(e => e.serviceName.toLowerCase().indexOf('spark') > -1)) {
+				this.endpoints.push(this.addCustomEndpoint(gatewayEndpoint, 'Spark History', '/gateway/default/sparkhistory'));
+			}
+			if (!endpointArray.find(e => e.serviceName.toLowerCase().indexOf('log') > -1)) {
+				this.endpoints.push(this.addCustomEndpoint(gatewayEndpoint, 'Yarn History', '/gateway/default/yarn'));
+			}
+		}
+	}
+
+	private isHyperlink(serviceName: string): boolean {
+		return !(serviceName.toLowerCase().indexOf(EndpointConstants.appproxy.toString()) > -1 || serviceName.toLowerCase().indexOf(EndpointConstants.controller.toString()) > -1
```

Comment:
This name doesn't seem accurate since it's checking for very specific links. Why not just do something more generic like look for http? Or at least make it clear what this is doing (either via doc comments or a better name)

Reply:
I'll add comments but the purpose I added it, is to check whether the endpoint is hyperlink or not. I know the implementation isn't checking for it in an obvious way, but for now it's what it is.